internal/installercache/installercache.go

```diff
@@ -177,41 +268,42 @@ func (i *Installers) evict() {
 			return nil
 		}
 
-		//save the other files based on their mod time
-		files.Add(&fileInfo{path, info})
-		totalSize += info.Size()
+		stat, ok := info.Sys().(*syscall.Stat_t)
+		if !ok || stat.Nlink == 1 {
+		    if !ok {
+  				i.log.Warnf("could not determine hardlink status for %s - item will not be filtered", path)
+  			}
```

Comment:
This indentation is weird, I imagine the linter will complain. You should probably run that locally before pushing.